src/shared/libs/helpers.ts

```diff
@@ -81,3 +81,7 @@ export function isEmptyArray<T>(arr: unknown): arr is T[] {
 export function isFullArray<T>(arr: unknown): arr is T[] {
   return Array.isArray(arr) && arr.length > 0;
 }
+
+export function isUndefined(value: unknown): value is undefined {
+  return typeof value === "undefined";
+}
```

Comment:
Why is this function needed when you can write `value === undefined`?

Reply:
`isUndefined` serves as a **type guard** in TypeScript, helping with type narrowing and improving code readability and consistency, especially in complex scenarios

While `value === undefined` works for simple checks, `isUndefined` enhances TypeScript’s type inference